worlds/smw/__init__.py

```diff
@@ -0,0 +1,250 @@
+        trap_weights += ([ItemName.literature_trap] * self.world.literature_trap_weight[self.player].value)
+        trap_count = 0 if (len(trap_weights) == 0) else math.ceil(junk_count * (self.world.trap_fill_percentage[self.player].value / 100.0))
+        junk_count -= trap_count
+
+        trap_pool = []
+        for i in range(trap_count):
+            trap_item = self.world.random.choice(trap_weights)
+            trap_pool += [self.create_item(trap_item)]
+
+        itempool += trap_pool
+
+        itempool += [self.create_item(ItemName.one_up_mushroom)] * junk_count
+
+        boss_location_names = [LocationName.yoshis_island_koopaling, LocationName.donut_plains_koopaling, LocationName.vanilla_dome_koopaling,
+                               LocationName.twin_bridges_koopaling, LocationName.forest_koopaling, LocationName.chocolate_koopaling,
+                               LocationName.valley_koopaling, LocationName.vanilla_reznor, LocationName.forest_reznor, LocationName.chocolate_reznor, LocationName.valley_reznor]
+
+        for location_name in boss_location_names:
+            self.world.get_location(location_name, self.player).place_locked_item(self.create_item(ItemName.koopaling))
+
+        self.world.itempool += itempool
+
+
+    def generate_output(self, output_directory: str):
+        try:
+            world = self.world
+            player = self.player
+
+            rom = LocalRom(get_base_rom_path())
+            patch_rom(self.world, rom, self.player, self.active_level_dict)
+
+            outfilepname = f'_P{player}'
+            outfilepname += f"_{world.player_name[player].replace(' ', '_')}" \
+                if world.player_name[player] != 'Player%d' % player else ''
+
+            rompath = os.path.join(output_directory, f'AP_{world.seed_name}{outfilepname}.sfc')
+            rom.write_to_file(rompath)
+            self.rom_name = rom.name
+
+            patch = SMWDeltaPatch(os.path.splitext(rompath)[0]+SMWDeltaPatch.patch_file_ending, player=player,
+                                  player_name=world.player_name[player], patched_path=rompath)
+            patch.write()
+        except:
+            raise
+        finally:
+            if os.path.exists(rompath):
+                os.unlink(rompath)
+            self.rom_name_available_event.set() # make sure threading continues and errors are collected
+
+    def modify_multidata(self, multidata: dict):
+        import base64
+        # wait for self.rom_name to be available.
+        self.rom_name_available_event.wait()
+        rom_name = getattr(self, "rom_name", None)
+        # we skip in case of error, so that the original error in the output thread is the one that gets raised
+        if rom_name:
+            new_name = base64.b64encode(bytes(self.rom_name)).decode()
+            multidata["connect_names"][new_name] = multidata["connect_names"][self.world.player_name[self.player]]
+
+        if self.topology_present:
```

Comment:
There is a dedicated hook for hint info now, since https://github.com/ArchipelagoMW/Archipelago/pull/1014